src/eos/radiationProperties/radiationZimmer.cpp

```diff
@@ -0,0 +1,241 @@
+    PetscReal YinCH4 = (functionContext->densityYiCH4Offset == -1) ? 0 : conserved[functionContext->densityYiCH4Offset] / density;
+    PetscReal YinCO = (functionContext->densityYiCOOffset == -1) ? 0 : conserved[functionContext->densityYiCOOffset] / density;
+
+    /** Computing the partial pressure of each species*/
+    pCO2 = (density * UGC * YinCO2 * temperature) / (MWCO2 * 101325.);
+    pH2O = (density * UGC * YinH2O * temperature) / (MWH2O * 101325.);
+    pCH4 = (density * UGC * YinCH4 * temperature) / (MWCH4 * 101325.);
+    pCO = (density * UGC * YinCO * temperature) / (MWCO * 101325.);
+
+    /** The resulting absorptivity is an average of species absorptivity weighted by partial pressure. */
+    *kappa = pCO2 * kappaCO2 + pH2O * kappaH2O + pCH4 * kappaCH4 + pCO * kappaCO;
+    PetscFunctionReturn(0);
+}
+PetscErrorCode ablate::eos::radiationProperties::Zimmer::ZimmerTemperatureFunction(const PetscReal *conserved, PetscReal temperature, PetscReal *kappa, void *ctx) {
+    PetscFunctionBeginUser;
+
+    /** This model depends on mass fraction, temperature, and density in order to predict the absorption properties of the medium. */
+    auto functionContext = (FunctionContext *)ctx;
+    double density;       //!< Variables to hold information gathered from the fields
+    PetscErrorCode ierr;  //!< Standard PETSc error code returned by PETSc functions
+
+    ierr = functionContext->densityFunction.function(conserved, &density, functionContext->densityFunction.context.get());  //!< Get the density value at this location
+    CHKERRQ(ierr);
+
+    double kapparef = 1;  //!< Reference absorptivity
+    double Tsurf = 300.;  //!< Reference temperature in Kelvin
+
+    /** The Zimmer model uses a fit approximation of the absorptivity. This depends on the presence of four species which are present in combustion and shown below. */
+    double kappaH2O = 0;
+    double kappaCO2 = 0;
+    double kappaCH4 = 0;
+    double kappaCO = 0;
+    double pCO2, pH2O, pCH4, pCO;
+
+    /** These are the constants stored for the model. */
+    std::array<double, 7> H2O_coeff[] = {{0.22317e1, -.15829e1, .1329601e1, -.50707, .93334e-1, -0.83108e-2, 0.28834e-3}};
+    std::array<double, 7> CO2_coeff[] = {{0.38041E1, -0.27808E1, 0.11672E1, -0.284910E0, 0.38163E-1, -0.26292E-2, 0.73662E-4}};  //!< polynomial coefficients for H2O
+    std::array<double, 5> CH4_coeff[] = {{6.6334E0, -3.5686E-3, 1.6682E-8, 2.5611E-10, -2.6558E-14}};                            //!< polynomial coefficients for CH4
+    std::array<double, 5> CO_1_coeff[] = {{4.7869E0, -6.953E-2, 2.95775E-4, -4.25732E-7, 2.02894E-10}};                          //!< polynomial coefficients for CO with T <= 750 K
+    std::array<double, 5> CO_2_coeff[] = {{1.0091E1, -1.183E-2, 4.7753E-6, -5.87209E-10, -2.5334E-14}};                          //!< polynomial coefficients for CO with T > 750 K
+    double MWC = 1.2010700e+01;
```

Comment:
Any of these constants that are used in both the functions should be defined once in the header file.